include/modules/meta/base.inl

```diff
@@ -249,31 +249,18 @@ namespace modules {
           non_tag = string_util::ltrim(move(non_tag), ' ');
         }
       }
+      m_builder->node(non_tag);
+
+      if (has_tags) {
+        // format-spacing is added between all tags
+        m_builder->spacing(format->spacing);
+      }
+      has_tags = true;
 
       string tag = value.substr(start, end - start + 1);
       bool tag_built = CONST_MOD(Impl).build(&tag_builder, tag);
       string tag_content = tag_builder.flush();
-
-      /*
-       * Remove exactly one space between two tags if the second tag was not built.
-       */
-      if (!tag_built && has_tags && !format->spacing) {
-        if (!non_tag.empty() && non_tag.back() == ' ') {
-          non_tag.erase(non_tag.size() - 1);
-        }
-      }
-
-      m_builder->node(non_tag);
-
-      if (tag_built) {
-        if (has_tags) {
-          // format-spacing is added between all tags
-          m_builder->spacing(format->spacing);
-        }
-
-        m_builder->node(tag_content);
-        has_tags = true;
-      }
+      m_builder->node(tag_built ? tag_content : tag);
```

Comment:
Unfortunately we can't rely on the return value of `build` for this, I should have mentioned this in the original issue because I already tried that.

There are some modules that return `false` for valid tags (e.g. the xkeyboard module for `label-indicator` if there are no indicators. In that case displaying `<label-indicator>` is an error.

Try the following xkeyboard module:

```dosini
[module/xkeyboard]
type = internal/xkeyboard
blacklist-0 = num lock
blacklist-1 = scroll lock
blacklist-2 = caps lock
format = <label-indicator>
label-layout = %name%
```

We have to check `format->tags` to see if a tag is valid.

Reply:
I see, I didn't catch that. In that case, it would be convenient to have `format->tags` be a set instead, or is there something that relies on it being a vector?

Reply:
I don't think anything else uses it. Feel free to use whatever data structure you want (or completely throw it away and do something else :wink:)